### DISCUSSION TITLE
PlayerAttackCooldownResetEvent

### DISCUSSION THREAD

**Discussion by Lukadcf** [PaperMC/Paper#11323] [Ideas]
### Is your feature request related to a problem?

The attack cooldown currently gets reset to 0.0 when a player attacks or switches held materials by scrolling/hotkey. I believe the latter is punishing the player for using items other than their weapon so I want to change this for my server.

### Describe the solution you'd like.

I would like a cancellable event that gets fired every time a player's attack cooldown resets to 0.0. Preferably with the cause of the event.

Example use:
```
@EventHandler
public void onAttackCooldown(PlayerAttackCooldownResetEvent event) {
Player player = event.getPlayer();
if (event.getCause() == PlayerAttackCooldownResetEvent.Cause.SWITCH_MATERIAL || player.isOp()) { 
event.setCancelled(true); 
   }
}
```

### Describe alternatives you've considered.

I tried using Player#resetCooldown() but this doesn't actually reset the player's attack indicator, and even if it did it wouldn't help me because I want the cooldown to continue where it left so not setting the cooldown at all makes the most sense.

I also tried calculating the GENERIC_ATTACK_SPEED a player needs to have for the cooldown to be as long as where it left off when swapping items, which kinda works but is very gimmicky and is 2 ticks delayed because players attribute instance takes that long to update after the PlayerItemHeldEvent

### Other

![image](https://github.com/user-attachments/assets/a1112449-8bd7-48eb-8922-4f8cb09fdce8)

**Comment by Machine-Maker** [comment]
Don't crosspost. You seem to have also requested this [as a feature upstream](https://hub.spigotmc.org/jira/browse/SPIGOT-7880). This just wastes everyone's time. Someone who sees this, spends their time implementing it, only to be superseded by upstream.

**Reply by Lukadcf** [reply]
Sorry, I'll close this as duplicate

> Don't crosspost. You seem to have also requested this [as a feature upstream](https://hub.spigotmc.org/jira/browse/SPIGOT-7880). This just wastes everyone's time. Someone who sees this, spends their time implementing it, only to be superseded by upstream.
